loop.py

```diff
@@ -112,8 +152,12 @@ def run_model(r, w, k, theta, D=20, iterations=350):
             print(f"{iteration} Ranking with a total cost of {m.objective_value}")
             print("Number of solutions: %d" % m.num_solutions)
 
-            # array when the value on the i_th position indicates the subject at rank i
+            # array with the value on the i_th position indicates the subject at rank i
             new_ranking = convert_solution_to_ranking(x)
+
+            if k > 1:
```

Comment:
This would never work if k==1 (and it should)

Reply:
I added this line as when k == 1 it means we're using the Uniform attention model, right? Our implementation for the probabilistic model is based on the Geometric model as I understood.